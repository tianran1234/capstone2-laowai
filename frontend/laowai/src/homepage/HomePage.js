import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "./HomePage.css";
import UserContext from "../auth/UserContext";

/** Homepage of site.
 *
 * Shows welcome message or login/register buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

function HomePage() {
  const { currentUser } = useContext(UserContext);
  console.debug("Homepage", "currentUser=", currentUser);

  
  return (
      <div className="Homepage">
        <div className="container text-center">
          <h1 className="mb-4 font-weight-bold">Welcome to LaoWai</h1>
          <p className="lead">Get connected and stay tuned.</p>
          {!currentUser
          ? (
            <p>
              <Link className="btn btn-primary font-weight-bold mr-3"
                    to="/auth/login">
                Log in
              </Link>
              <Link className="btn btn-primary font-weight-bold"
                    to="/auth/signup">
                Sign up
              </Link>
            </p>
          ) : <h2>
                Welcome Back, {currentUser.firstName || currentUser.username}!
              </h2>
  
        }
      </div>
    </div>
  );
}



export default HomePage;
