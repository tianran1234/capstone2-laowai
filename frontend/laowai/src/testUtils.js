import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  image_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
