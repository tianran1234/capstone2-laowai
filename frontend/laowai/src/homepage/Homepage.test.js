import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import HomePage from "./HomePage";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <HomePage />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider currentUser={null}>
          <HomePage />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
