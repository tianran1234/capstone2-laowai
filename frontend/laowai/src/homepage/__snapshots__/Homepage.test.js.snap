// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Welcome to LaoWai
      </h1>
      <p
        class="lead"
      >
        Get connected and stay tuned.
      </p>
      <h2>
        Welcome Back, testuser!
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Welcome to LaoWai
      </h1>
      <p
        class="lead"
      >
        Get connected and stay tuned.
      </p>
      <p>
        <a
          class="btn btn-primary font-weight-bold mr-3"
          href="/auth/login"
        >
          Log in
        </a>
        <a
          class="btn btn-primary font-weight-bold"
          href="/auth/signup"
        >
          Sign up
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
